### PR TITLE
workflows: Run demo workflow

### PR DESCRIPTION
**Pull request by vladkanatov**
- Add run-demo.py checks workflow

### CODE REVIEW

File: .github/workflows/run-demo.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    
+
+jobs:
+  demo:
+    name: Run demo 
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout demo repository
+        uses: actions/checkout@v3
```
Comment (adityasaky) [inline .github/workflows/run-demo.yml:19]:
```suggestion
        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683
```

File: .github/workflows/run-demo.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        with:
+          repository: gittuf/demo
+
+      - name: Install Python
+        uses: actions/setup-python@v4
```
Comment (adityasaky) [inline .github/workflows/run-demo.yml:24]:
```suggestion
        uses: actions/setup-python@0b93645e9fea7318ecaed2b359559ac225c90a2b
```

File: .github/workflows/run-demo.yml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        with:
+          python-version: '3.11'
```
Comment (adityasaky) [inline .github/workflows/run-demo.yml:26]:
```suggestion
          python-version: '3.12'
```

**Review by adityasaky — CHANGES_REQUESTED** [review]
Thanks @vladkanatov, this looks good. Can you pin the actions to hashes + bump the python version?

**Comment by vladkanatov** [comment]
Okay👌

**Pull request merged**